src/java/fr/paris/lutece/plugins/suggest/business/Category.java

```diff
@@ -100,12 +100,8 @@ public void setTitle( String title )
      */
     public boolean equals( Object obj )
     {
-        if ( obj instanceof Category && ( (Category) obj ).getIdCategory( ) == _nIdCategory )
-        {
-            return true;
-        }
+        return obj != null && obj == this.getClass() && ( (Category) obj ).getIdCategory( ) == _nIdCategory;
```

Comment:
this seems wrong. `obj` is almost certainly not the class `Category`

Reply:
Ok, I'll rollback to "obj instanceof Category" instead of "this.getClass()".